components/hab/src/cli.rs

```diff
@@ -900,6 +900,14 @@ pub fn get(feature_flags: FeatureFlag) -> App<'static, 'static> {
     )
 }
 
+fn maybe_subcommand<T: StructOpt>(condition: bool) -> App<'static, 'static> {
+    if condition {
+        T::clap()
+    } else {
+        App::new("")
```

Comment:
This is determined at the call site. For [example](https://github.com/habitat-sh/habitat/blob/57430a72c543bfa83c1149c2a613b1d65b628db4/components/hab/src/cli.rs#L834), this will add the `SvcBulkLoad` subcommand if the `SERVICE_CONFIG_FILES` feature flag is set. Otherwise, it adds a completely empty app. This is a bit hacky, but it was the best I could come up with. The compile-time nature of `structopt` makes it difficult to conditionally add subcommands or arguments at runtime.